packages/presentation/presentation_publishing/interfaces/has_compressed.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+export interface HasCompressed {
+  compressed: boolean;
+}
+
+export const apiHasCompressed = (unknownApi: unknown): unknownApi is HasCompressed => {
+  return (
+    (unknownApi as HasCompressed).compressed !== undefined &&
+    typeof (unknownApi as HasCompressed).compressed === 'function'
```

Comment:
Type should verify 'boolean', not 'function'. Also, lets put this in controls plugin, since only controls uses this interface for now